### ISSUE TITLE
Use scikit-build for PyPI package?

### ISSUE THREAD

**Issue by cmichelenstrofer** [capytaine/capytaine#115]
I have not been able to install capytaine from PyPI using `pip` on my mac (default Fortran compiler does not work and haven't figured out a workaround). It installs with `conda` seamlessly. 

It seems like `scikit-build` would do for `pip` many of the things that `conda` does in the background to make Fortran compilation easier. [See this article.](https://python-packaging-tutorial.readthedocs.io/en/latest/binaries_dependencies.html) 
It might be worth exploring building the PyPI package with [scikit-build](https://scikit-build.readthedocs.io/en/latest/).

**Comment by mancellin** [comment]
I agree that it would be nice to improve the `pip` packaging of Capytaine.
Thank you for the reference, I was not aware of `scikit-build`, I'll look into it.

**Comment by cmichelenstrofer** [comment]
FYI for those of us using Macs:
 
Installing Capytaine requires compiling fortran. When installing with `pip` you need to make sure you have fortran compilers. **_Macs do not come with a Fortran compiler_**. 

**Install gfortran** 
`gfortran` is part of the GNU compiler Collection `gcc`. There are several ways to install it, here is how I did it using MacPorts:

1. install xcode (large) from Apple
2. install command-line-tools for xcode from Apple
3. install MacPorts
4. Install & Activate gcc (and gfortran) using MacPorts
```bash
sudo port install gcc11 +gfortran
sudo port select --set gcc mp-gcc11
```
5. Add ``export CC="gcc"`` to your `.zshrc` file

**Comment by kmruehl** [comment]
@mancellin I"m also having issues with the FORTRAN compiler, but on Windows. I'm trying to pip install from GitHub, and am getting a `no Fortran compiler found` error.

**Comment by mancellin** [comment]
There used to be [instructions in the doc on how to set up a fortran compiler on Windows](https://github.com/mancellin/capytaine/blob/v1.1/docs/developer_manual/installation.rst). But this is a bit cumbersome and I now recommend installing the precompiled binaries from `conda` for users and using Windows Subsystem for Linux for developpers.
Having precompiled binaries also on `pip` is on my roadmap, but it is a fair amount of work (and not the fun kind of work).

**Comment by kmruehl** [comment]
@mancellin thanks. I was able to do a conda installation on Windows. A pip install from GitHub is my preference, so thank you for pointing me to the prior developer installation instructions (I couldn't find them).

**Comment by mancellin** [comment]
> thank you for pointing me to the prior developer installation instructions (I couldn't find them).

@kmruehl If it works for you, I could add them back to the documentation. I removed them last year because I wasn't sure they were correct and up-to-date, and I did not have the time to double check them.

**Comment by cmichelenstrofer** [comment]
Another improvement is to switch to the new `pyproject.toml` instead of `setup.py` since this will be deprecated soon.

**Comment by cmichelenstrofer** [comment]
Also, The installation instructions above (for Macs) is not working with Python 3.10. I'll stick to 3.9 for now. Hopefully using scikit-build will fix this.

**Comment by mancellin** [comment]
Here is the current status:

- This thread might be closed by PR #335 since it mostly answer the original suggestion. Capytaine is currently built with `numpy.distutils` which will be removed from the next version of Python. There are two main possibilities to replace it: `scikit-build` based on [CMake](https://cmake.org/) and `meson-python` based on [Meson](https://mesonbuild.com/). I've chosen to try the latter, since Meson feels a bit user-friendlier than CMake. `Meson-python` is currently used to build Scipy and considered to build Numpy and Pandas, so I'm relatively confident in its long term support.

- I took the opportunity also to switch to `pyproject.toml` in #331 as suggested by @cmichelenstrofer.

- This change will affect how the package is built on Windows. It may or may not become easier I haven't tested that yet.

- In the procees of researching about `scikit-build` and `meson-python`, I found out about [cibuildwheels](https://cibuildwheel.readthedocs.io) that is a tool to automate the packaging for `pip` in a CI workflow such as Github Actions. I'm experimenting with it in #329. I'm hoping to deliver a precompiled package on `pip` for the next release of Capytaine.
